aws-blog-emrencyrption/cfn/kms.json

```diff
@@ -0,0 +1,240 @@
+                                                "arn:aws:iam::",
+                                                {
+                                                    "Ref": "AWS::AccountId"
+                                                },
+                                                ":role/EMR_DefaultRole"
+                                            ]
+                                        ]
+                                    }
+                                ]
+                            },
+                            "Resource": [
+                                "*"
+                            ],
+                            "Sid": "Stmt1"
+                        },
+                        {
+                            "Action": [
+                                "kms:Put*",
+                                "kms:ScheduleKeyDeletion",
+                                "kms:CancelKeyDeletion",
+                                "kms:Describe*",
+                                "kms:Revoke*",
+                                "kms:Disable*",
+                                "kms:Enable*",
+                                "kms:Delete*",
+                                "kms:List*",
+                                "kms:Update*",
+                                "kms:Create*"
+                            ],
+                            "Effect": "Allow",
+                            "Principal": {
+                                "AWS": [
+                                    {
+                                        "Fn::Join": [
+                                            "",
+                                            [
+                                                "arn:aws:iam::",
+                                                {
+                                                    "Ref": "AWS::AccountId"
+                                                },
+                                                ":root"
+                                            ]
+                                        ]
+                                    }
+                                ]
+                            },
+                            "Resource": [
+                                "*"
+                            ],
+                            "Sid": "Stmt2"
+                        }
+                    ],
+                    "Version": "2012-10-17"
+                }
+            },
+            "Type": "AWS::KMS::Key"
+        },
+        "KMSLUKSKey": {
+            "Properties": {
+                "Description": "Master Key that will be used for LUKS Encryption",
+                "Enabled": "true",
+                "KeyPolicy": {
+                    "Statement": [
+                        {
+                            "Action": [
+                                "kms:Encrypt",
+                                "kms:Decrypt",
+                                "kms:ReEncrypt",
```

Comment:
for LUKS, the permission that we need here are: 
"kms:Decrypt",
 "kms:GenerateDataKey",